File: ios/StillMoment/Presentation/Views/GuidedMeditations/PlayerCenterDisc.swift
//
//  PlayerCenterDisc.swift
//  Still Moment
//
//  Presentation Layer — statische Gluehscheibe als visueller Anker im Player.
//

import SwiftUI

/// Statische Gluehscheibe hinter dem Pause-Button im Atemkreis-Player.
///
/// Kein Skalieren, kein Pulsieren, kein Opazitaets-Wechsel — reine visuelle
/// Ruhezone, die den zentralen Bereich des Rings warm anhebt. Im Dark Mode
/// leicht waermer und sichtbarer, im Light Mode dezenter. Hex-Werte und
/// Opacities folgen dem KS-2.0-Player-Handoff.
struct PlayerCenterDisc: View {
    @Environment(\.colorScheme)
    private var colorScheme

    var body: some View {
        Circle()
            .fill(
                RadialGradient(
                    gradient: Gradient(stops: self.gradientStops),
                    center: .center,
                    startRadius: 0,
                    endRadius: 110
                )
            )
            .frame(width: 220, height: 220)
            .allowsHitTesting(false)
    }

    private var gradientStops: [Gradient.Stop] {
        switch self.colorScheme {
        case .dark:
            [
                Gradient.Stop(
                    color: Color(red: 214.0 / 255, green: 138.0 / 255, blue: 110.0 / 255, opacity: 0.10),
                    location: 0.0
                ),
                Gradient.Stop(
                    color: Color(red: 199.0 / 255, green: 125.0 / 255, blue: 99.0 / 255, opacity: 0.04),
                    location: 0.5
                ),
                Gradient.Stop(color: .clear, location: 0.8),
                Gradient.Stop(color: .clear, location: 1.0)
            ]
        default:
            [
                Gradient.Stop(
                    color: Color(red: 162.0 / 255, green: 80.0 / 255, blue: 62.0 / 255, opacity: 0.07),
                    location: 0.0
                ),
                Gradient.Stop(
                    color: Color(red: 162.0 / 255, green: 80.0 / 255, blue: 62.0 / 255, opacity: 0.03),
                    location: 0.5
                ),
                Gradient.Stop(color: .clear, location: 0.8),
                Gradient.Stop(color: .clear, location: 1.0)
            ]
        }
    }
}

// MARK: - Previews

@available(iOS 17.0, *)
#Preview("Dark") {
    ZStack {
        LinearGradient(
            colors: [
                Color(red: 0.10, green: 0.06, blue: 0.04),
                Color(red: 0.20, green: 0.12, blue: 0.10),
                Color(red: 0.36, green: 0.23, blue: 0.18)
            ],
            startPoint: .top,
            endPoint: .bottom
        )
        .ignoresSafeArea()

        PlayerCenterDisc()
    }
    .preferredColorScheme(.dark)
}

@available(iOS 17.0, *)
#Preview("Light") {
    ZStack {
        LinearGradient(
            colors: [
                Color(red: 0.98, green: 0.93, blue: 0.86),
                Color(red: 0.97, green: 0.80, blue: 0.66),
                Color(red: 0.91, green: 0.63, blue: 0.46)
            ],
            startPoint: .top,
            endPoint: .bottom
        )
        .ignoresSafeArea()

        PlayerCenterDisc()
    }
    .preferredColorScheme(.light)
}

File: android/app/src/main/kotlin/com/stillmoment/presentation/ui/meditations/components/PlayerCenterDisc.kt
package com.stillmoment.presentation.ui.meditations.components

import androidx.compose.foundation.Canvas
import androidx.compose.foundation.isSystemInDarkTheme
import androidx.compose.foundation.layout.size
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import com.stillmoment.presentation.ui.theme.StillMomentTheme

private val DEFAULT_SIZE = 220.dp

// Hex-Werte 1:1 aus iOS' PlayerCenterDisc.swift (KS-2.0-Player-Handoff).
// Im Dark Mode leicht waermer und sichtbarer, im Light Mode dezenter.
private val DiscWarmDark = Color(0xFFD68A6E) // rgb(214, 138, 110)
private val DiscWarmDarkMid = Color(0xFFC77D63) // rgb(199, 125, 99)
private val DiscWarmLight = Color(0xFFA2503E) // rgb(162, 80, 62)

private const val DARK_ALPHA_INNER = 0.10f
private const val DARK_ALPHA_MID = 0.04f
private const val LIGHT_ALPHA_INNER = 0.07f
private const val LIGHT_ALPHA_MID = 0.03f

/**
 * Statische Gluehscheibe hinter dem Pause-Button im Player-Ring (shared-096).
 *
 * Kein Skalieren, kein Pulsieren, kein Opazitaets-Wechsel — reine visuelle
 * Ruhezone, die den zentralen Bereich des Rings warm anhebt. Im Dark Mode
 * leicht waermer und sichtbarer, im Light Mode dezenter. Hex-Werte und
 * Opacities folgen dem KS-2.0-Player-Handoff, 1:1 vom iOS-Pendant.
 *
 * Dekoration — kein `contentDescription`/`semantics`-Block (Compose-Default:
 * `Box`/`Canvas` ohne Semantics-Modifier sind nicht im a11y-Tree).
 */
@Composable
fun PlayerCenterDisc(modifier: Modifier = Modifier, size: Dp = DEFAULT_SIZE) {
    val isDark = isSystemInDarkTheme()

    val stops = if (isDark) {
        arrayOf(
            0.0f to DiscWarmDark.copy(alpha = DARK_ALPHA_INNER),
            0.5f to DiscWarmDarkMid.copy(alpha = DARK_ALPHA_MID),
            0.8f to Color.Transparent,
            1.0f to Color.Transparent,
        )
    } else {
        arrayOf(
            0.0f to DiscWarmLight.copy(alpha = LIGHT_ALPHA_INNER),
            0.5f to DiscWarmLight.copy(alpha = LIGHT_ALPHA_MID),
            0.8f to Color.Transparent,
            1.0f to Color.Transparent,
        )
    }

    Canvas(modifier = modifier.size(size)) {
        val radius = this.size.minDimension / 2f
        drawCircle(
            brush = Brush.radialGradient(
                colorStops = stops,
                center = center,
                radius = radius,
            ),
            radius = radius,
            center = center,
        )
    }
}

// MARK: - Previews

@Preview(name = "Disc - Dark", showBackground = true, backgroundColor = 0xFF1A100A)
@Composable
private fun PlayerCenterDiscDarkPreview() {
    StillMomentTheme {
        PlayerCenterDisc()
    }
}

@Preview(name = "Disc - Light", showBackground = true, backgroundColor = 0xFFFFE3D6)
@Composable
private fun PlayerCenterDiscLightPreview() {
    StillMomentTheme {
        PlayerCenterDisc()
    }
}
